Engine/Graphics/OpenGL/RenderOpenGL.cpp

```diff
@@ -1445,44 +1117,49 @@ void RenderOpenGL::BlendTextures(int x, int y, Image* imgin, Image* imgblend, in
 //_4A65CC(unsigned int x, unsigned int y, Texture_MM7 *a4, Texture_MM7 *a5, int a6, int a7, int a8)
 // a6 is time, a7 is 0, a8 is 63
 void RenderOpenGL::TexturePixelRotateDraw(float u, float v, Image *img, int time) {
-    // TODO(pskelton): sort this - forcing the draw is slow - precalculate/ shader
+    // TODO(pskelton): sort this - precalculate/ shader
+    static Texture *cachedtemp[14]{};
+    static int cachetime[14]{ -1 };
 
     if (img) {
-        uint8_t *palpoint24 = (uint8_t *)img->GetPalette();
-        int width = img->GetWidth();
-        int height = img->GetHeight();
-        Texture *temp = CreateTexture_Blank(width, height, IMAGE_FORMAT_A8R8G8B8);
-        uint32_t *temppix = (uint32_t *)temp->GetPixels(IMAGE_FORMAT_A8R8G8B8);
-
-        uint8_t *texpix24 = (uint8_t *)img->GetPixels(IMAGE_FORMAT_R8G8B8);
-        uint8_t thispix;
-        int palindex;
-
-        for (int dy = 0; dy < height; ++dy) {
-            for (int dx = 0; dx < width; ++dx) {
-                thispix = *texpix24;
-                if (thispix >= 0 && thispix <= 63) {
-                    palindex = (time + thispix) % (2 * 63);
-                    if (palindex >= 63)
-                        palindex = (2 * 63) - palindex;
-                    temppix[dx + dy * width] = Color32(palpoint24[palindex * 3], palpoint24[palindex * 3 + 1], palpoint24[palindex * 3 + 2]);
+        std::string_view tempstr{ *img->GetName() };
+        int number = tempstr[4] - 48;
```

Comment:
Wow that's some wizardry. Is there no other way to do this?

Reply:
Yes this is a very hacky fix/workaround to stop having to force the drawcall every image. I intend to sort a better solution at some point.

Reply:
👍

Reply:
Let's add a TODO then